tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Remoting/Server/ProcessMessageIntegration.cs

```diff
@@ -0,0 +1,111 @@
+        MethodName = "ProcessMessage",
+        ReturnTypeName = "System.Runtime.Remoting.Channels.ServerProcessing",
+        ParameterTypeNames = new[] { "System.Runtime.Remoting.Channels.IServerChannelSinkStack", "System.Runtime.Remoting.Messaging.IMessage", "System.Runtime.Remoting.Channels.ITransportHeaders", "System.IO.Stream", "System.Runtime.Remoting.Messaging.IMessage&", "System.Runtime.Remoting.Channels.ITransportHeaders&", "System.IO.Stream&", },
+        MinimumVersion = RemotingIntegration.Major4,
+        MaximumVersion = RemotingIntegration.Major4,
+        IntegrationName = RemotingIntegration.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    // ReSharper disable once InconsistentNaming
+    public class ProcessMessageIntegration
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProcessMessageIntegration));
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of target</typeparam>
+        /// <typeparam name="TServerSinkStack">Type of the server sink stack</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="sinkStack">Server sink stack instance</param>
+        /// <param name="requestMsg">The incoming request message instance</param>
+        /// <param name="requestHeaders">The headers for the incoming request message</param>
+        /// <param name="requestStream">The stream for the incoming request message</param>
+        /// <param name="responseMsg">The outgoing response message instance</param>
+        /// <param name="responseHeaders">The headers for the outgoing response message</param>
+        /// <param name="responseStream">The stream for the outgoing response message</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TServerSinkStack>(TTarget instance, TServerSinkStack sinkStack, IMessage requestMsg, ITransportHeaders requestHeaders, Stream requestStream, ref IMessage responseMsg, ref ITransportHeaders responseHeaders, ref Stream responseStream)
+        {
+            if (requestMsg is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            // Extract span context
+            SpanContext propagatedContext = null;
+            try
+            {
+                propagatedContext = SpanContextPropagator.Instance.Extract(requestHeaders, (headers, key) =>
+                {
+                    var value = headers[key];
+                    if (value is null)
+                    {
+                        return new string[] { };
+                    }
+                    else
+                    {
+                        return new string[] { value.ToString() };
+                    }
```

Comment:
Should use `Array.Empty<string>()` here. The ternery is optional 😉 
```suggestion
                    return value is null
                        ? Array.Empty<string>()
                        : new string[] { value.ToString() };
```

Reply:
Fixed in 424990f39cd3c504c09b7978f938c653e225a365